core/src/test/java/com/amplifyframework/core/model/query/predicate/OperatorTest.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.core.model.query.predicate;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests operator evaluation.
+ */
+@SuppressWarnings("MagicNumber")
+public final class OperatorTest {
+
+    /**
+     * Test the accuracy of Equal operator evaluation.
+     */
+    @Test
+    public void testEqualOperator() {
+        final EqualQueryOperator numericalOperator = new EqualQueryOperator(123);
+
+        Assert.assertTrue(numericalOperator.evaluate(123));
+        Assert.assertFalse(numericalOperator.evaluate(1234));
+        Assert.assertFalse(numericalOperator.evaluate("123"));
+
+        final EqualQueryOperator stringOperator = new EqualQueryOperator("Hello");
+
+        Assert.assertTrue(stringOperator.evaluate("Hello"));
+        Assert.assertFalse(stringOperator.evaluate("HELLO"));
+        Assert.assertFalse(stringOperator.evaluate("World"));
+    }
+
+    /**
+     * Test the accuracy of Not Equal operator evaluation.
+     */
+    @Test
+    public void testNotEqualOperator() {
+        final NotEqualQueryOperator numericalOperator = new NotEqualQueryOperator(123);
+
+        Assert.assertTrue(numericalOperator.evaluate(124));
+        Assert.assertTrue(numericalOperator.evaluate("123"));
```

Comment:
Ah, gee. Is the `String` `"123"` equal to the `int` or `Integer` `123?` I don't know. This looks a conscious design decision, more than a black-and-white kind of thing. But, [by default, anyway, they __are not__ equal](https://ideone.com/f47Fbt).

Reply:
Personally as a user, I would be more interested in the comparison of values rather than the technicality of data type. For example, it would be very inconvenient to have to check if `Person.age` is stored as `Integer` vs `int` before doing a predicate statement.

Reply:
Oh I thought you were talking about comparing `int 123` and `new Integer(123)`, which should be equal. `String "123"` and `int 123` should NOT be equal as you say, and the test operator is using `NotEqualQueryOperator`, so it will evaluate to true when compared.